lib/treewalk/walk.go

```diff
@@ -0,0 +1,57 @@
+package treewalk
+
+import (
+	"errors"
+)
+
+type Node interface {
+	NextChild() Node
+}
+
+func Walk(root Node, preVisit WalkFunc, postVisit WalkFunc) error {
+	return walk(root, preVisit, postVisit)
+}
```

Comment:
Wow that's... a really good question.  I think because this evolved by mutation from `filepath.Walk` and that had a bunch of internal complications that made starting different than the rest of traversal, and I got rid of that in this implementation.  Removing.